### ISSUE TITLE
error creating database

### ISSUE THREAD

**Issue by kussmaul** [Lunatic-Labs/rubricapp#359]
I cloned the code (to my Windows laptop and to a Fedora Core 36 VM in VMware), and tried to follow the instructions in `README.md`, but had problems.

`pip install -r requirements.txt` produced this message (in Fedora):
```
    Defaulting to user installation because normal site-packages is not writeable
```
and everything seemed to install correctly in ` c:\users\kussmaul\appdata\roaming\python\python311\site-packages` (Windows) or `~/.local/lib/python3.10/site-packages` (Fedora).

`python dbcreate.py .` produced this error (on both platforms, with appropriately different file paths):
```
Traceback (most recent call last):
  File "/home/kussmaul/Documents/rubricapp/dbcreate.py", line 3, in <module> db.create_all() 
  File "/home/kussmaul/.local/lib/python3.10/site-packages/flask_sqlalchemy/extension.py", line 868, in create_all
    self._call_for_binds(bind_key, "create_all")
  File "/home/kussmaul/.local/lib/python3.10/site-packages/flask_sqlalchemy/extension.py", line 839, in _call_for_binds
    engine = self.engines[key]
  File "/home/kussmaul/.local/lib/python3.10/site-packages/flask_sqlalchemy/extension.py", line 628, in engines
    app = current_app._get_current_object()  # type: ignore[attr-defined]
  File "/home/kussmaul/.local/lib/python3.10/site-packages/werkzeug/local.py", line 513, in _get_current_object
    raise RuntimeError(unbound_message) from None
RuntimeError: Working outside of application context.

This typically means that you attempted to use functionality that needed the current application. 
To solve this, set up an application context with app.app_context(). See the documentation for more information.
```

Any suggestions for what went wrong, or what I should do next? Are there other, better, installation instructions?
Thanks,
Clif

**Comment by brianlugo1** [comment]
Greetings @kussmaul, currently there is a pending pull request that has updated documentation in the README.md and updated changes that fix the error produced when running dbcreate.py!

**Comment by mekealbrown** [comment]
@kussmaul , Pull request has been merged into master.

**Comment by kussmaul** [comment]
@brianlugo1 and @mekealbrown, **thank you** for the quick response.
TL;DR - I was able to run the dev server on both Windows & Fedora.

Details: On both systems (Windows laptop and Fedora VM), I did the following:
  1. checked that `python` and `pip` were Python 3
  1. `git pull` to get current code
  1. `pip install -r requirements.txt` in case requirements had changed
    -  on both systems, ran with no problems
  1. `python setupEnv.py`
    - on both, I got a warning that `account.db` didn't exist so couldn't be removed
    - on Fedora, the dev server ran. **Success!**
    - on Windows, I got errors:
```
$ python setupEnv.py .
rm: cannot remove 'C:\Users\Kussmaul\Documents\Data\projects\rubricapp\core\account.db': No such file or directory
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.
```

Looking at the code, `setupEnv.py` calls `python3`. On my Windows laptop, `python` is defined but `python3` isn't. I ran `dbcreate.py` and `run.py` with `python`, and the dev server ran. **Success!**

Is it worth the time & effort to make `setupEnv.py` more robust? It might be tricky given the various ways Python can be installed on Windows.

**Comment by brianlugo1** [comment]
@kussmaul The pull request that I made has just been merged! Can you let me know if you are still getting the errors on your Windows machine?

**Comment by kussmaul** [comment]
@brianlugo1 sorry for the delay. Today I tried again from scratch (on Windows).
I deleted Python 2 since it caused path problems and I don't think I need it. I deleted & reinstalled Python 3 to have a clean start, using Chocolatey, which I use for many programs. In Windows PowerShell, I tried:
```
choco install python3 # as Administrator
python --version # Python 3.11.0
pip    --version # pip 22.3
git clone https://github.com/kussmaul/rubricapp
cd rubricapp
pip install -r requirements.txt # as Administrator (see below)
python setupEnv.py
```

When I ran `pip install` as a regular user, I got deprecation warnings for `visitor`, `fpdf`, `MarkupSafe`, and `flask_bootstrap`, and an error: 
```
WARNING: Failed to write executable - trying to use .deleteme logic
ERROR: Could not install packages due to an OSError: [WinError 2] The system cannot find the file specified: 
'C:\\Python311\\Scripts\\sqlformat.exe' -> 'C:\\Python311\\Scripts\\sqlformat.exe.deleteme'
```

When I ran `pip install` as Administrator, I still get the 4 deprecation warnings, but no errors.
The deprecation warnings all look like this:
```
DEPRECATION: visitor is being installed using the legacy 'setup.py install' method, 
because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. 
pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. 
Discussion can be found at https://github.com/pypa/pip/issues/8559
```

When I ran `python setupEnv.py` (without the "." argument) everything seemed to work, and I see:
```
Could Not Find rubricapp\core\account.db
The system cannot find the file specified.
 * Serving Flask app 'core'
 * Debug mode: on
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:5000
Press CTRL+C to quit
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 420-350-213
127.0.0.1 - - [28/Dec/2022 12:49:46] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [28/Dec/2022 12:49:46] "GET /static/bootstrap.css HTTP/1.1" 200 -
127.0.0.1 - - [28/Dec/2022 12:49:46] "GET /static/cover.css HTTP/1.1" 200 -
127.0.0.1 - - [28/Dec/2022 12:49:47] "GET /favicon.ico HTTP/1.1" 404 -
```

Suggestions:
- In README, mention the `sqlformat` error and deprecation warnings, since I'm not sure we can (or should) do much about them.
- In README, indicate which commands should run as Administrator/superuser.
- In README, describe `python run.py` and remove it from `setupEnv.py`, which I would expect to setup the environment, and not to run the dev server.

OK by me to close this issue, since the original problem seems to be solved by the PRs. Thank you!

**Comment by sah0017** [comment]
@brianlugo1 Would you update the README with the items that Clif described here?
